app/controllers/api/host_initiators_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Api
+  class HostInitiatorsController < BaseController
+    def refresh_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" if id.blank?
+
+      ensure_resource_exists(type, id) if single_resource?
+
+      api_action(type, id) do |klass|
+        host_initiator = resource_search(id, type, klass)
+        api_log_info("Refreshing #{host_initiator_ident(host_initiator)}")
+        refresh_host_initiator(host_initiator)
+      end
+    end
+
+    def create_resource(_type, _id = nil, data = {})
+      raise BadRequestError, "ems_id not defined for #{_type} resource" if data['ems_id'].blank?
```

Comment:
Interesting, we don't do this for any other EMS lookups by `date["ems_id"]`, I would have expected the `.find()` to raise an exception and the API to return an appropriate error but this works too.
@abellotti is this a pattern you'd like to follow more broadly?

We're doing similar lookups here without this check:
```
app/controllers/api/auth_key_pairs_controller.rb:      ext_management_system = ExtManagementSystem.find(data['ems_id'])
app/controllers/api/cloud_tenants_controller.rb:      ext_management_system = resource_search(data['ems_id'], :providers, collection_class(:providers))
app/controllers/api/physical_storages_controller.rb:      ext_management_system = ExtManagementSystem.find(data['ems_id'])
app/controllers/api/host_aggregates_controller.rb:      ext_management_system = resource_search(data['ems_id'], :providers, collection_class(:providers))
app/controllers/api/floating_ips_controller.rb:      ems = ExtManagementSystem.find(data['ems_id'])
app/controllers/api/cloud_volumes_controller.rb:      ext_management_system = ExtManagementSystem.find(data['ems_id'])
```